### ISSUE TITLE
add tests for logging of unhandled deferred errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2463]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#2463|
|Type|enhancement|
|Created|2007-02-20 11:36:31Z|

Playing around with the way deferreds are GC'ed I wrote a few more tests to check that unhandled errors are properly logged.
One of the tests is commented out since it fails (because of a deliberate ref cycle that prevents the DebugInfo to be GC'ed).

Attachments:

* [unhandled-deferred-tests.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/db9/db9af2f9bf14fd96af0c66e8193df7b24a72e6b8/8e6f02f93dd446293ab0315e2b33d9504e3e46d3.patch) (2995 bytes) - added by antoine on 2007-02-20 11:37:32Z - patch for test_defer.py

<details><summary>Searchable metadata</summary>

```
trac-id__2463 2463
type__enhancement enhancement
reporter__antoine antoine
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1171971391000000 1171971391000000
changetime__1174160319000000 1174160319000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Ready to review in defer-gc-2463.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @therve|
|-|-|

If ```test_errorLog``` is going to get a docstring, it should get an accurate docstring.  It should say something like
```
Verify that when a Deferred with no references to it is fired, and its final
result (the one not handled by any callback) is an exception, that exception
will be logged immediately.
```
Also, the commented-out block can be removed, I don't think that is explaining anything useful to anybody.

Aside from those minor tweaks, this looks good.  Go ahead and merge when you have applied them.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [19849]) Merge defer-gc-2463

Author: antoine, therve
Reviewer: glyph
Fixes [#2463](https://github.com/twisted/twisted/issues/2463)

Add tests to ensure unhandled errors from Deferred are correctly logged.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
